edk2toollib/windows/capsule/inf_generator2.py

```diff
@@ -0,0 +1,399 @@
+        InfStrings.AddLocalizableString("MfgName", Manufacturer)
+
+    @property
+    def Name(self):
+        return self._name
+
+    @Name.setter
+    def Name(self, value):
+        # test here for invalid chars
+        if not (re.compile(r'[\w-]*$')).match(value):
+            raise ValueError("Name has invalid chars.")
+        self._name = value
+
+    @property
+    def VersionStr(self):
+        return self._versionstr
+
+    @VersionStr.setter
+    def VersionStr(self, value):
+        c = value.count(".")
+        if(c < 1) or (c > 3):
+            raise ValueError("VersionString must be in format of xx.xx -> xx.xx.xx.xx")
+        self._versionstr = value
+
+    @property
+    def Date(self):
+        return self._date
+
+    @Date.setter
+    def Date(self, value):
+        self._date = datetime.strptime(value, "%m/%d/%Y").strftime("%m/%d/%Y")
+
+    @property
+    def Arch(self):
+        return self._arch
+
+    @Arch.setter
+    def Arch(self, value):
+        key = value.lower()
+        if(key not in SUPPORTED_ARCH.keys()):
+            raise ValueError("Unsupported Architecture")
+        self._arch = SUPPORTED_ARCH[key]
+
+    def __str__(self) -> str:
+        '''Return the string representation of this InfHeader object'''
+        return textwrap.dedent(f"""\
+            ;
+            ; {self.Name}
+            ; {self.VersionStr}
+            ; Copyright (C) Microsoft Corporation. All Rights Reserved.
+            ;
+            [Version]
+            Signature="$WINDOWS NT$"
+            Class=Firmware
+            ClassGuid={{f2e7dd72-6468-4e36-b6f1-6488f42c1b52}}
+            Provider=%Provider%
+            DriverVer={self.Date},{self.VersionStr}
+            PnpLockdown=1
+            CatalogFile={self.Name}.cat
+
+            [Manufacturer]
+            %MfgName% = Firmware,NT{self.Arch}
+
+            """)
+
+
+class InfFirmware(object):
+    def __init__(self, Tag: str, Description: str, EsrtGuid: str, VersionInt: str, FirmwareFile: str,
+                 InfStrings: 'InfStrings', InfSourceFiles: 'InfSourceFiles', Rollback: bool = False,
+                 IntegrityFile: str = None) -> None:
+        '''Instantiate an INF firmware object.
+        This object represents individual firmware sections within the INF.
+
+        Tag             - A string that uniquely identifies this firmware (e.g. "Firmware0")
+        Description     - A description of the firmware (e.g. "UEFI Firmware")
+        EsrtGuid        - ESRT GUID for this firmware in string format. (e.g. "34e094e9-4079-44cd-9450-3f2cb7824c97")
+        VersionInt      - Version as an integer in string format (e.g. "1234" or "0x04158000")
+        FirmwareFile    - Filename (basename only) of the firmware payload file (e.g. "Firmware1234.bin")
+        InfStrings      - An InfStrings object representing the "Strings" section of this INF file.
+        InfSourceFiles  - An InfSourceFIles object representing the "SourceDisks*" sections of this INF file.
```

Comment:
InfSourceFIles should be InfSourceFiles.  Notice the I in Files

Reply:
Fixed.